source/js/buyers-guide/search/search-filter.js

```diff
@@ -21,6 +24,27 @@ export class SearchFilter {
 
     this.allProducts = document.querySelectorAll(`figure.product-box`);
     this.categoryTitle = document.querySelector(`.category-title`);
+
+    // we want the animation to start when the first eight products images are loaded
+    Promise.all(
+      Array.from(
+        document.querySelectorAll(".product-box.d-flex img.product-thumbnail")
+      )
+        .slice(0, 8)
+        .filter((img) => !img.complete)
+        .map(
+          (img) =>
+            new Promise((resolve) => {
+              img.onload = img.onerror = resolve;
```

Comment:
surely this should be `img.onload = resolve; img.onerror = reject`?

(and then using [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) rather than `Promise.all`)